src/donation/models/donation.entity.ts

```diff
@@ -0,0 +1,35 @@
+import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
+import { UserEntity } from '../../user/models/user.entity';
+
+@Entity('donations')
+export class DonationEntity {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @ManyToOne(() => UserEntity, user => user.donations)
+  user_: UserEntity;
+
+  @Column({ nullable: true })
+  companion_user_id: number;
+
+  @Column()
+  donated_type: string;
+
+  @Column()
+  amount: number;
+
+  @Column({ nullable: true })
+  blood_pressure: string;
+
+  @Column({ nullable: true })
+  hemoglobin: number;
+
+  @Column({ nullable: true })
+  details: string;
+
+  @Column({ type: 'date' })
```

Comment:
This could actually be `datetime` - in case we wanted to run any events/quests based on the time of donation (e.g. early bird for donating in the morning)

Reply:
Good point, will correct.